### ISSUE TITLE
Simplify function transform a string to number, which cause some problems.

### ISSUE THREAD

**Issue by HK-SHAO** [josdejong/mathjs#2152]
Why simplify function transform a string to number? I don't want to change the typeof value, because it may cause some problems.

```javascript
math.simplify('f("0xffff")')
// args[0]: r {value: 123}

math.simplify('"1234"')
// {value: 1234}
```

**Comment by josdejong** [comment]
That's a good point, I see `simplify` does turn strings into a number when it contains a numeric value.

Makes sense to me to remove this behavior.

Anyone able to dive into this?

**Comment by HK-SHAO** [comment]
> That's a good point, I see `simplify` does turn strings into a number when it contains a numeric value.
> 
> Makes sense to me to remove this behavior.
> 
> Anyone able to dive into this?

I don't think it's a good behavior. I have read the source code of simplify and want to try to modify it. Later I gave up because it is very difficult for me. 🤣

**Comment by josdejong** [comment]
Hm ok, thanks for giving it a try at least 👍

**Comment by josdejong** [comment]
Fix is published now in `v10.1.0`, see #2372
